rolling-deploy.sh

```diff
@@ -92,27 +136,31 @@ echo "Performing zero-downtime cutover to ${APP_NAME}-${BUILD_NUMBER}"
 cf map-route "${APP_NAME}-${BUILD_NUMBER}" "${CF_EXTERNAL_APPS_DOMAIN}" -n "${EXTERNAL_APP_HOSTNAME}"
 
 echo "A/B deployment"
-if [[ ! -z "${DEPLOYED_APP}" && "${DEPLOYED_APP}" != "" ]]; then
+if [[ -n "${DEPLOYED_APP}" && -n "${TARGET_INSTANCES}" ]]; then
 
     declare -i instances=0
-    declare -i old_app_instances=${DEPLOYED_APP_INSTANCES}
-    echo "begin scaling down from: ${DEPLOYED_APP_INSTANCES}"
-
-    while (( ${instances} != ${DEPLOYED_APP_INSTANCES} )); do
-      	declare -i instances=${instances}+1
-		declare -i old_app_instances=${old_app_instances}-1
-      	echo "Scaling up ${APP_NAME}-${BUILD_NUMBER} to ${instances}.."
-      	cf scale -i ${instances} "${APP_NAME}-${BUILD_NUMBER}"
-
-		  echo "Scaling down ${DEPLOYED_APP} to ${old_app_instances}.."
-		  cf scale -i ${old_app_instances} "${DEPLOYED_APP}"
+    declare -i old_app_instances=${TARGET_INSTANCES}
+    echo "begin scaling down from: ${TARGET_INSTANCES}"
+
+    while ((${instances} != ${TARGET_INSTANCES})); do
+        declare -i instances=${instances}+1
+        declare -i old_app_instances=${old_app_instances}-1
+        echo "Scaling up ${APP_NAME}-${BUILD_NUMBER} to ${instances}.."
```

Comment:
you using ${APP_NAME}-${BUILD_NUMBER} multiple times so might make sense to store to local variable